src/geo/ui/legends/legends-view.js

```diff
@@ -21,10 +21,8 @@ var LegendsView = View.extend({
   },
 
   _initBinds: function () {
-    this._layersCollection.on('add remove', this._onLayerAddedOrRemoved, this);
-    this.add_related_model(this._layersCollection);
-    this.settingsModel.on('change', this._onSettingsModelChanged, this);
-    this.add_related_model(this.settingsModel);
+    this.listenTo(this._layersCollection, 'add remove layerMoved', this._onLegendsChanged);
+    this.listenTo(this.settingsModel, 'change', this._onSettingsModelChanged);
```

Comment:
Don't we have to make `settingsModel` required? If it is not present, it might fail.